### PR TITLE
Refactor for dynamic_imdb plugin

### PR DESCRIPTION
**Pull request by liiight**
Complete refactor, more efficient and readable.
Major bug fix when selecting a female person (yeah, seriously)

Added ability to use multiple IDs

### CODE REVIEW

File: flexget/plugins/input/dynamic_imdb.py [hunk 1]
```diff
@@ -112,130 +138,126 @@ def get_entity_type_and_object(self, imdb_id):
...
-                            if item not in items and item['kind'] in content_types:
-                                log.debug('Found item: ' + item.get('title') + ', adding to unfiltered list')
-                                items.append(item)
-                            else:
-                                log.debug('Item ' + str(item) + ' already found in list, skipping.')
-        return items
+                yield el
 
-    def get_items_by_character(self, character, content_types):
+    def no_empty_objects(self, _list):
```
Comment (gazpachoking) [inline flexget/plugins/input/dynamic_imdb.py]: There's a builtin for this, `filter(None, thelist)` (the first argument is a discriminator function, but if you pass None it just filters items that aren't truthy.)

File: flexget/plugins/input/dynamic_imdb.py [hunk 2]
```diff
@@ -112,130 +138,126 @@ def get_entity_type_and_object(self, imdb_id):
...
+        flat_list = self.flatten_list(non_flat_list)
+        if remove_none:
+            flat_list = self.no_empty_objects(flat_list)
+        return flat_list
+
+    def filtered_items(self, unfiltered_items, content_types):
         items = []
-        for content_type in content_types:
-            items += character.get(CHARACTER_CONTENT_TYPE_CONVERSION[content_type], [])
+        unfiltered_items = list(set(unfiltered_items))
```
Comment (gazpachoking) [inline flexget/plugins/input/dynamic_imdb.py]: No need to turn this back into a list here, `sorted` can operate on any iterable.

File: flexget/plugins/input/dynamic_imdb.py [hunk 3]
```diff
@@ -112,130 +138,126 @@ def get_entity_type_and_object(self, imdb_id):
...
+        if entity_type == 'Company':
+            return self.items_by_company(entity_object)
+
+        if entity_type == 'Character':
+            return self.items_by_character(entity_object, content_types)
+
+        elif entity_type == 'Person':
+            return self.items_by_person(entity_object, job_types, content_types)
+
+    def flatten_list(self, _list):
```
Comment (gazpachoking) [inline flexget/plugins/input/dynamic_imdb.py]: There is [itertools.chain](https://docs.python.org/2/library/itertools.html#itertools.chain) available from the stdlib for this.

Comment (liiight) [inline flexget/plugins/input/dynamic_imdb.py]: I remember trying this but something didn't work right. Is using collections an issue?

Comment (liiight) [inline flexget/plugins/input/dynamic_imdb.py]: Trying to switch to using chain turns out to be a real pain and requires some more changes. If it's not a big deal, I'd like to remain with the current method :)

**Pull request merged**